src/hotspot/share/opto/graphKit.cpp

```diff
@@ -619,6 +619,40 @@ void GraphKit::builtin_throw(Deoptimization::DeoptReason reason, Node* arg) {
     }
   }
 
+  if (treat_throw_as_hot && OptimizeImplicitExceptions) {
+    ciInstanceKlass* ex_ciInstKlass = env()->exception_instanceKlass_for_reason(reason);
+    if (ex_ciInstKlass != NULL) {
+      const TypeKlassPtr *ex_type = TypeKlassPtr::make(ex_ciInstKlass);
+      kill_dead_locals();
+      Node* ex_node = new_instance(makecon(ex_type), NULL, NULL, true);
```

Comment:
What happened if deoptimization happen during this allocation (which is safepoint)?
Which bytecode will be executed in Interpeter after deopt?